src/amo/reducers/addonsByAuthors.js

```diff
@@ -1,125 +1,146 @@
+  type: typeof FETCH_ADDONS_BY_AUTHORS,
+  payload: FetchAddonsByAuthorsParams,
 |};
 
-export const fetchOtherAddonsByAuthors = (
-  { addonType, authors, errorHandlerId, slug }: FetchOtherAddonsByAuthorsParams
-): FetchOtherAddonsByAuthorsAction => {
-  if (!errorHandlerId) {
-    throw new Error('An errorHandlerId is required');
-  }
-
-  if (!slug) {
-    throw new Error('An add-on slug is required.');
-  }
-
-  if (!addonType) {
-    throw new Error('An add-on type is required.');
-  }
-
-  if (!authors) {
-    throw new Error('Authors are required.');
-  }
-
-  if (!Array.isArray(authors)) {
-    throw new Error('The authors parameter must be an array.');
-  }
+export const fetchAddonsByAuthors = (
+  { addonType, authors, errorHandlerId, forAddonSlug }: FetchAddonsByAuthorsParams
+): FetchAddonsByAuthorsAction => {
+  invariant(errorHandlerId, 'An errorHandlerId is required');
+  invariant(addonType, 'An add-on type is required.');
+  invariant(authors, 'Authors are required.');
+  invariant(Array.isArray(authors), 'The authors parameter must be an array.');
 
   return {
-    type: FETCH_OTHER_ADDONS_BY_AUTHORS,
+    type: FETCH_ADDONS_BY_AUTHORS,
     payload: {
       addonType,
       authors,
       errorHandlerId,
-      slug,
+      forAddonSlug,
     },
   };
 };
 
-type LoadOtherAddonsByAuthorsParams = {|
-  slug: string,
+type LoadAddonsByAuthorsParams = {|
   addons: Array<ExternalAddonType>,
+  forAddonSlug?: string,
 |};
 
-type LoadOtherAddonsByAuthorsAction = {|
-  type: typeof LOAD_OTHER_ADDONS_BY_AUTHORS,
-  payload: LoadOtherAddonsByAuthorsParams,
+type LoadAddonsByAuthorsAction = {|
+  type: typeof LOAD_ADDONS_BY_AUTHORS,
+  payload: LoadAddonsByAuthorsParams,
 |};
 
-export const loadOtherAddonsByAuthors = (
-  { addons, slug }: LoadOtherAddonsByAuthorsParams
-): LoadOtherAddonsByAuthorsAction => {
-  if (!slug) {
-    throw new Error('An add-on slug is required.');
-  }
-
-  if (!addons) {
-    throw new Error('A set of add-ons is required.');
-  }
+export const loadAddonsByAuthors = (
+  { addons, forAddonSlug }: LoadAddonsByAuthorsParams
+): LoadAddonsByAuthorsAction => {
+  invariant(addons, 'A set of add-ons is required.');
 
   return {
-    type: LOAD_OTHER_ADDONS_BY_AUTHORS,
-    payload: { slug, addons },
+    type: LOAD_ADDONS_BY_AUTHORS,
+    payload: { addons, forAddonSlug },
   };
 };
 
 type Action =
-  | FetchOtherAddonsByAuthorsAction
-  | LoadOtherAddonsByAuthorsAction;
+  | FetchAddonsByAuthorsAction
+  | LoadAddonsByAuthorsAction;
 
 const reducer = (
   state: State = initialState,
   action: Action
 ): State => {
   switch (action.type) {
-    case FETCH_OTHER_ADDONS_BY_AUTHORS:
-      return {
-        ...state,
-        byAddonSlug: {
-          ...state.byAddonSlug,
-          [action.payload.slug]: undefined,
-        },
-      };
-    case LOAD_OTHER_ADDONS_BY_AUTHORS:
+    case FETCH_ADDONS_BY_AUTHORS:
+      if (!action.payload.forAddonSlug) {
+        return state;
```

Comment:
You still need to reset the `byUserId` / `byUserName` state by making empty arrays for each author that is being fetched, right? Otherwise it might be hard to render a loading state.

Reply:
Yup, this was a lazy first patch. Sorry, I've been rushing patches out lately; feeling a bit behind. I'll get better at that 😓

Reply:
I think it's fine to start with incomplete patches. A todo comment to fix it later would have been fine. It's always easier to review smaller patches!